tutorials/pretraining-Vietnamese-data-curation/pretraining-Vietnamese-data-curation.ipynb

```diff
@@ -0,0 +1,655 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites and Environment setups"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "-\tCUDA and NVIDIA Drivers: CUDA 12.3 with Driver 545.23.08\n",
+    "-\tUbuntu 22.04\n",
+    "-\t[NVIDIA-container-toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html) version 1.15.0\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Install NeMo Curator by following the instructions to install the CPU and CUDA-accelerated modules in the README file of the [NeMo Curator repository](https://github.com/NVIDIA/NeMo-Curator/tree/main). "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, install additional packages that will be needed later, which are datasets and jsonlines.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install datasets\n",
+    "!pip install jsonlines"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To proceed with data processing, we need to set up a Dask environment. Dask is a flexible, open-source library that enables parallel and distributed computing in Python, allowing us to scale computations across multiple cores or even clusters. By distributing tasks, Dask makes the data handling process significantly faster and more efficient."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Note:** This notebook was run on a DGX A100 with a 128-core CPU and 2TB of RAM to handle the dataset size. Depending on your dataset and computing resources, you may need to adjust the Dask worker configuration (in the next section) accordingly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nemo_curator\n",
+    "from dask.distributed import Client, LocalCluster\n",
+    "# Start a Dask cluster with 12 workers, each limited at 64GB of memory. You might need to adjust these numbers according to your computing resources\n",
+    "cluster = LocalCluster(n_workers=12, processes=True, memory_limit= '64GB')\n",
+    "client = Client(cluster)\n"
+   ]
```

Comment:
I tried it, but in the parameters of the get_client function, I didn't set the 'memory_limit' parameter because this function doesn't support it. If I don't set this parameter, the server will crash if the notebook runs into an out-of-memory (OOM) issue.

Reply:
Ah okay, thanks for verifying! Opened https://github.com/NVIDIA/NeMo-Curator/issues/324 for the NeMo Curator team to fix this eventually. For now, having it this way in the tutorial is fine.

Reply:
Actually, can you share which version of NeMo Curator you are using? This should have been fixed with https://github.com/NVIDIA/NeMo-Curator/pull/266.